geonode/base/api/views.py

```diff
@@ -717,11 +720,17 @@ def resource_service_ingest(self, request, resource_type: str = None):
             return Response(status=status.HTTP_403_FORBIDDEN)
         try:
             request_params = QueryDict(request.body, mutable=True)
+            uuid = request_params.get('uuid', str(uuid1()))
+            try:
+                resource = ResourceBase.objects.get(uuid=uuid)
```

Comment:
Generally speaking this is not the most efficient and clean way to look for an object on the DB. I would remove this `Exception` and use the `filer.exists` or `filter.count` in order to check if there's a resource on the DB.